src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/AttachAddressToBuildingUnitLambdaHandler.cs

```diff
@@ -0,0 +1,81 @@
+namespace BuildingRegistry.Api.BackOffice.Handlers.Lambda.Handlers.BuildingUnit
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Abstractions.Validation;
+    using Be.Vlaanderen.Basisregisters.AggregateSource;
+    using Be.Vlaanderen.Basisregisters.Api.Exceptions;
+    using Be.Vlaanderen.Basisregisters.Sqs.Exceptions;
+    using Be.Vlaanderen.Basisregisters.Sqs.Lambda.Handlers;
+    using Be.Vlaanderen.Basisregisters.Sqs.Lambda.Infrastructure;
+    using Be.Vlaanderen.Basisregisters.Sqs.Responses;
+    using BuildingRegistry.Api.BackOffice.Abstractions.Building.Validators;
+    using BuildingRegistry.Building;
+    using BuildingRegistry.Building.Exceptions;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Configuration;
+    using Requests.BuildingUnit;
+    using TicketingService.Abstractions;
+
+    public sealed class AttachAddressToBuildingUnitLambdaHandler : BuildingUnitLambdaHandler<AttachAddressToBuildingUnitLambdaRequest>
+    {
+        public AttachAddressToBuildingUnitLambdaHandler(
+            IConfiguration configuration,
+            ICustomRetryPolicy retryPolicy,
+            ITicketing ticketing,
+            IIdempotentCommandHandler idempotentCommandHandler,
+            IBuildings buildings)
+            : base(
+                configuration,
+                retryPolicy,
+                ticketing,
+                idempotentCommandHandler,
+                buildings)
+        { }
+
+        protected override async Task<ETagResponse> InnerHandle(AttachAddressToBuildingUnitLambdaRequest request, CancellationToken cancellationToken)
+        {
+            var cmd = request.ToCommand();
+
+            try
+            {
+                await IdempotentCommandHandler.Dispatch(
+                    cmd.CreateCommandId(),
+                    cmd,
+                    request.Metadata,
+                    cancellationToken);
+            }
+            catch (IdempotencyException)
+            {
+                // Idempotent: Do Nothing return last etag
+            }
+
+            var lastHash = await GetHash(
```

Comment:
Shouldn't we be adding the relation building unit - address to a back office context as in parcel registry?

Reply:
+1